cgen: fn as arg of a method

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
```
OS: macos, macOS, 12.3.1, 21E258
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

getwd: /Users/hunam/Documents/dev/vweb
vmodules: /Users/hunam/.vmodules
vroot: /Applications/v
vexe: /Applications/v/v
vexe mtime: 2022-05-18 14:31:59
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 ac90a2b.5b96f7e

Git version: git version 2.32.1 (Apple Git-133)
Git vroot status: weekly.2022.20-47-ge57080f3 (9 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-unknown-unknown de82a130
```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
struct Foo {}

fn (f Foo) bar(baz fn ()) {}

fn baz() {}

fn main() {
	foo := Foo{}
	foo.bar(baz())
}
```

**What did you see instead?**
```c
/tmp/v_501/report.12276697961892973920.tmp.c:2077:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/report.12276697961892973920.tmp.c:11555:30: error: operand of type 'void' where arithmetic or pointer type is required
        main__Foo_bar(foo, (voidptr)main__baz());
                                    ^~~~~~~~~~~
1 warning and 1 error generated.
```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1463,10 +1463,8 @@ pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 						continue
 					}
 				}
-				if got_arg_typ != ast.void_type {
-					c.error('$err.msg() in argument ${i + 1} to `${left_sym.name}.$method_name`',
-						arg.pos)
-				}
+				c.error('$err.msg() in argument ${i + 1} to `${left_sym.name}.$method_name`',
+					arg.pos)
 			}
 		}
 		if method.is_unsafe && !c.inside_unsafe {

--- a/vlib/v/checker/tests/method_call_arg_mismatch.out
+++ b/vlib/v/checker/tests/method_call_arg_mismatch.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/method_call_arg_mismatch.vv:9:10: error: `baz()` (no value) used as value in argument 1 to `Foo.bar`
+    7 | fn main() {
+    8 |     foo := Foo{}
+    9 |     foo.bar(baz())
+      |             ~~~~~
+   10 | }

--- a/vlib/v/checker/tests/method_call_arg_mismatch.vv
+++ b/vlib/v/checker/tests/method_call_arg_mismatch.vv
@@ -1,0 +1,10 @@
+struct Foo {}
+
+fn (f Foo) bar(baz fn ()) {}
+
+fn baz() {}
+
+fn main() {
+	foo := Foo{}
+	foo.bar(baz())
+}

